src/build/__init__.py

```diff
@@ -161,6 +163,17 @@ def __init__(self, srcdir='.', config_settings=None, python_executable=sys.execu
             python_executable=python_executable,
         )
 
+    @property
+    def python_executable(self):  # type: () -> Union[bytes, Text]
+        # make mypy happy
+        if not isinstance(self.hook.python_executable, typing.Text) and not isinstance(self.hook.python_executable, bytes):
+            raise TypeError('python_executable is not a string: {}'.format(type(self.hook.python_executable)))
+        return self.hook.python_executable
```

Comment:
If it's just to make mypy happy, this would work:

```suggestion
        python_executable = self.hook.python_executable  # type: Union[bytes, Text]
        return python_executable
```

And then you can revert to returning `self.hook.python_executable` directly whenever pep517 adds type hints.

Reply:
Sure. But the current approach works and does not cost us very much.